apps/condo/domains/billing/utils/countrySpecificQRCodeParsers/ru.qrCodeParser.js

```diff
@@ -0,0 +1,39 @@
+const iconv = require('iconv-lite')
+const { get } = require('lodash')
+
+/**
+ * @typedef {Object} TRUQRCodeFields
+ * @property {string} BIC
+ * @property {string} PayerAddress
+ * @property {string} PaymPeriod
+ * @property {string} Sum
+ * @property {string} PersAcc The resident's account
+ * @property {string} PayeeINN
+ * @property {string} PersonalAcc The bank account of the receiver
+ */
+
+/**
+ * @param {string} qrStr The QR code string got from the picture
+ * @return {TRUQRCodeFields}
+ */
+function parseRUReceiptQRCode (qrStr) {
+    const matches = /^ST(?<version>\d{4})(?<encodingTag>\d)\|(?<requisitesStr>.*)$/g.exec(qrStr)
+
+    if (!matches) {
+        throw new Error('Invalid QR code')
+    }
+
+    const requisitesStr = get(matches, ['groups', 'requisitesStr'], '')
+
+    // https://encoding.spec.whatwg.org/#koi8-r
+    // https://encoding.spec.whatwg.org/#windows-1251
+    const encodingTag = get(matches, ['groups', 'encodingTag'])
+    const encoding = get({ 1: 'cp1251', 2: 'utf-8', 3: 'koi8-r' }, encodingTag, 'utf-8')
+
+    // Skip decoding for utf string
+    const decodedRequisitesStr = encodingTag === '2' ? requisitesStr : iconv.decode(requisitesStr, encoding)
+
+    return Object.fromEntries(decodedRequisitesStr.split('|').map((part) => part.split('=', 2)))
```

Comment:
I really cant figure out what's going on here

Reply:
This
```
field1=value1|field2=value2|...
```
transformed to
```
{ field1: 'value1', field2: 'value2', ... }
```